src/trackinfoobject.cpp

```diff
@@ -110,26 +101,49 @@ void TrackInfoObject::setDeleteOnReferenceExpiration(bool deleteOnReferenceExpir
     m_bDeleteOnReferenceExpiration = deleteOnReferenceExpiration;
 }
 
-void TrackInfoObject::setMetadata(const Mixxx::TrackMetadata& trackMetadata) {
-    // TODO(XXX): This involves locking the mutex for every setXXX
-    // method. We should figure out an optimization where there are private
-    // setters that don't lock the mutex.
-    setArtist(trackMetadata.getArtist());
-    setTitle(trackMetadata.getTitle());
-    setAlbum(trackMetadata.getAlbum());
-    setAlbumArtist(trackMetadata.getAlbumArtist());
-    setYear(trackMetadata.getYear());
-    setGenre(trackMetadata.getGenre());
-    setComposer(trackMetadata.getComposer());
-    setGrouping(trackMetadata.getGrouping());
-    setComment(trackMetadata.getComment());
-    setTrackNumber(trackMetadata.getTrackNumber());
-    setTrackTotal(trackMetadata.getTrackTotal());
-    setChannels(trackMetadata.getChannels());
-    setSampleRate(trackMetadata.getSampleRate());
-    setDuration(trackMetadata.getDuration());
-    setBitrate(trackMetadata.getBitrate());
-    setReplayGain(trackMetadata.getReplayGain());
+void TrackInfoObject::setTrackMetadata(
+        const Mixxx::TrackMetadata& trackMetadata,
+        QImage *pCoverArt,
+        bool parsedFromFile) {
+    {
+        // enter locking scope
+        QMutexLocker lock(&m_qMutex);
+
+        bool modified = compareAndSet(&m_bHeaderParsed, parsedFromFile);
+        bool modifiedReplayGain =
+                (m_metadata.getReplayGain() != trackMetadata.getReplayGain());
+        bool modifiedCoverArt = false;
+        if (m_metadata != trackMetadata) {
+            const Mixxx::ReplayGain replayGainBackup(m_metadata.getReplayGain());
```

Comment:
Looks like legacy code that survived the introduction of the ReplayGain 
class. Deleted.

On 02/28/2016 10:15 PM, Daniel Schürmann wrote:
>
> In src/trackinfoobject.cpp 
> <https://github.com/mixxxdj/mixxx/pull/900#discussion_r54356482>:
>
> > -    setBitrate(trackMetadata.getBitrate());
> > -    setReplayGain(trackMetadata.getReplayGain());
> > +void TrackInfoObject::setTrackMetadata(
> > +        const Mixxx::TrackMetadata& trackMetadata,
> > +        QImage *pCoverArt,
> > +        bool parsedFromFile) {
> > +    {
> > +        // enter locking scope
> > +        QMutexLocker lock(&m_qMutex);
> > +
> > +        bool modified = compareAndSet(&m_bHeaderParsed, parsedFromFile);
> > +        bool modifiedReplayGain =
> > +                (m_metadata.getReplayGain() != trackMetadata.getReplayGain());
> > +        bool modifiedCoverArt = false;
> > +        if (m_metadata != trackMetadata) {
> > +            const Mixxx::ReplayGain replayGainBackup(m_metadata.getReplayGain());
>
> Why we need a special treatment for replay gain here? Don't we trust the 
> file replay gain?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/900/files#r54356482>.
>